src/hyperlink/_url.py

```diff
@@ -827,14 +884,16 @@ def __init__(self, scheme=None, host=None, path=(), query=(), fragment=u'',
         return
 
     def get_decoded_url(self, lazy=False):
+        # type: (bool) -> DecodedURL
         try:
             return self._decoded_url
         except AttributeError:
-            self._decoded_url = DecodedURL(self, lazy=lazy)
+            self._decoded_url = DecodedURL(self, lazy=lazy)  # type: DecodedURL
```

Comment:
I think it is really weird that this type hint is needed; seems like a `mypy` bug.